test/integration/cli-domain-package-server-script/cli-domain-package-server-script.js

```diff
@@ -0,0 +1,378 @@
+              expect(compiledContent).to.not.contain('user');
+              done();
+            } catch(e) {
+              done(e);
+            }
+          }
+        );
+      });
+    });
+
+    describe('when component does require an npm module', function(){
+      var serverContent = 'var _ =require(\'underscore\');'
+        + '\nvar user = {name:\'John\'};\nmodule.exports.data=function(context,cb){'
+        + '\nreturn cb(null, _.has(user, \'name\'));\n};';
+
+      beforeEach(function(done){
+        fs.writeFileSync(path.resolve(componentPath, serverName), serverContent);
+        done();
+      });
+
+      it('should save compiled data provider', function(done){
+        var dependencies = {underscore: '1.8.3'};
+
+        packageServerScript(
+          {
+            componentPath: componentPath,
+            ocOptions: {
+              files: {
+                data: serverName
+              }
+            },
+            publishPath: publishPath,
+            webpack: webpackOptions,
+            dependencies: dependencies
+          },
+          function(err, res){
+            if (err) {
+              return done(err);
+            }
+            try {
+              expect(res.type).to.equal('node.js');
+              expect(res.src).to.equal('server.js');
+
+              var compiledContent = fs.readFileSync(path.resolve(publishPath, res.src), {encoding: 'utf8'});
+              expect(res.hashKey).to.equal(hashBuilder.fromString(compiledContent));
+              done();
+            } catch(e) {
+              done(e);
+            }
+          }
+        );
+      });
+
+      describe('end required depenencies is not present in the package.json', function(){
+        it('should throw an error with details', function(done){
+          var dependencies = {lodash: '1.0.0'};
+
+          packageServerScript(
+            {
+              componentPath: componentPath,
+              ocOptions: {
+                files: {
+                  data: serverName
+                }
+              },
+              publishPath: publishPath,
+              webpack: webpackOptions,
+              dependencies: dependencies
+            },
+            function(err, res){
+              expect(err).to.not.be.null;
+              expect(err).to.contain('Missing dependencies from package.json => "underscore"');
+              done();
+            }
+          );
+        });
+      });
+    });
+
+    describe('when component does require a relative path from an npm module', function(){
+      var serverContent = 'var data=require(\'react-dom/server\');module.exports.data=function(context,cb){return cb(null,data); };';
+
+      beforeEach(function(done){
+        fs.writeFileSync(path.resolve(componentPath, serverName), serverContent);
+        done();
+      });
+
+      it('should throw an error when the dependency is not present in the package.json', function(done){
```

Comment:
to keep consistency, can we do a describe for "when the dependency is not present in the package.json" and then a nested it for "should throw an error"?